### ISSUE TITLE
Access to Identity 'us-east-1:c48bc499-25e4-xxxx-xxxx-868edaaed70c' is forbidden.

### ISSUE THREAD

**Issue by JaidynBelbin** [aws/aws-sdk-android#175]
Okay, I'm really stumped with this problem, and haven't found any solution anywhere else on the web. 

I'm integrating Google Sign-In with my Android app, and am currently at the point of retrieving a Google token using the 'old' way specified by the Amazon docs (in contrast to the way Google says to retrieve an ID Token in their docs [here](https://developers.google.com/identity/sign-in/android/backend-auth) which is another question altogether. But I figured I should stick with Amazons' way, as I haven't found anyone else using a different way.

The code I use is here:

```
private class GetAndSetGoogleToken extends AsyncTask<Void, Void, String> {

        @Override
        protected String doInBackground(Void... params) {

            try {

                GooglePlayServicesUtil.isGooglePlayServicesAvailable(getApplicationContext());

                AccountManager am = AccountManager.get(getApplicationContext());
                Account[] accounts = am.getAccountsByType(GoogleAuthUtil.GOOGLE_ACCOUNT_TYPE);

                token = GoogleAuthUtil.getToken(getApplicationContext(), accounts[0].name,
                        "audience:server:client_id:" + serverClientId);

            } catch(GoogleAuthException ex) {
                Log.d(TAG, "GoogleAuthException has been thrown by GetAndSetGoogleToken!");
            } catch(IOException ex2) {
                Log.d(TAG, "IOException has been thrown by GetAndSetGoogleToken!");
            }

            return token;
        }

        @Override
        protected void onPostExecute(String token) {

            // Passing the token to the credentialsProvider
            setCredentials(token);

            super.onPostExecute(token);
        }
    }
```

As you can see in the `onPostExecute` method, I pass the token to my CognitoCachingCredentialsProvider, using this code:

```
private void setCredentials(String token) {

        Map<String, String> logins = new HashMap<>();
        logins.put("accounts.google.com", token);
        credentialsProvider.withLogins(logins);
    }
```

The `serverClientId` passed to the `GoogleAuthUtil.getToken()` method is one of the two client ID's I received when I created the project in the Google Console. The other one was a client ID for **Android** which I put in the Authenticated Providers section in my Amazon Cognito Identity Pool. Perhaps this is the problem? I'm not sure; I've tried every combination of the two I can think of.

As far as I can tell, I have followed the Amazon docs and related blog posts to the letter, but continually get a `NotAuthorizedError`. The full stack trace is below:

```
07-18 21:05:08.552 2889-3052/com.example.yardenbourg.mindhivepodcast E/AndroidRuntime: FATAL EXCEPTION: AsyncTask #2
                                                                                       Process: com.example.yardenbourg.mindhivepodcast, PID: 2889
                                                                                       java.lang.RuntimeException: An error occured while executing doInBackground()
                                                                                           at android.os.AsyncTask$3.done(AsyncTask.java:300)
                                                                                           at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:355)
                                                                                           at java.util.concurrent.FutureTask.setException(FutureTask.java:222)
                                                                                           at java.util.concurrent.FutureTask.run(FutureTask.java:242)
                                                                                           at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231)
                                                                                           at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
                                                                                           at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
                                                                                           at java.lang.Thread.run(Thread.java:841)
                                                                                        Caused by: com.amazonaws.services.cognitoidentity.model.NotAuthorizedException: Access to Identity 'us-east-1:c48bc499-25e4-4af6-a2ee-868edaaed70c' is forbidden. (Service: AmazonCognitoIdentity; Status Code: 400; Error Code: NotAuthorizedException; Request ID: d5f18fbb-4d4c-11e6-ae3e-8d2ec33632dc)
                                                                                           at com.amazonaws.http.AmazonHttpClient.handleErrorResponse(AmazonHttpClient.java:716)
                                                                                           at com.amazonaws.http.AmazonHttpClient.executeHelper(AmazonHttpClient.java:392)
                                                                                           at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:199)
                                                                                           at com.amazonaws.services.cognitoidentity.AmazonCognitoIdentityClient.invoke(AmazonCognitoIdentityClient.java:533)
                                                                                           at com.amazonaws.services.cognitoidentity.AmazonCognitoIdentityClient.getCredentialsForIdentity(AmazonCognitoIdentityClient.java:406)
                                                                                           at com.amazonaws.auth.CognitoCredentialsProvider.populateCredentialsWithCognito(CognitoCredentialsProvider.java:627)
                                                                                           at com.amazonaws.auth.CognitoCredentialsProvider.startSession(CognitoCredentialsProvider.java:553)
                                                                                           at com.amazonaws.auth.CognitoCredentialsProvider.refresh(CognitoCredentialsProvider.java:503)
                                                                                           at com.amazonaws.auth.CognitoCachingCredentialsProvider.refresh(CognitoCachingCredentialsProvider.java:463)
                                                                                           at com.example.yardenbourg.mindhivepodcast.MindHiveLogin$RefreshCredentials.doInBackground(MindHiveLogin.java:154)
                                                                                           at com.example.yardenbourg.mindhivepodcast.MindHiveLogin$RefreshCredentials.doInBackground(MindHiveLogin.java:149)
                                                                                           at android.os.AsyncTask$2.call(AsyncTask.java:288)
                                                                                           at java.util.concurrent.FutureTask.run(FutureTask.java:237)
                                                                                           at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231) 
                                                                                           at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112) 
                                                                                           at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587) 
                                                                                           at java.lang.Thread.run(Thread.java:841) 
```

I would really, really appreciate some help with this. I cannot find anyone else having the same problem as me anywhere. Thanks in advance.

**Comment by JaidynBelbin** [comment]
Never mind, I simply needed to modify my Trust Policies to include "accounts.google.com" as a trusted entity. Now it works perfectly.

**Comment by KdB13** [comment]
I'm having the same problem using **KInesis Video Streams**. When I sign in for the first time, it works. However, when I reopen my app I get this `NotAuthorizedException`. I've also rechecked my **Pool ID** in the `awsconfiguration.json`.

I followed the official docs for the [Android Producer Library](https://docs.aws.amazon.com/kinesisvideostreams/latest/dg/producer-sdk-android.html) and created the Identity Pool and set the policy as mentioned in the docs.
